### ISSUE TITLE
[Feature Request] Import via Exchange API

### ISSUE THREAD

**Issue by flying-sausages** [Xtrendence/Cryptofolio#45]
I have not yet spun an instance up but I haven't seen a mention of this on the README.

I'd love to have my transactions imported from my exchanges instead of doing it all manually. The CFolio API is exposed already which might allow for this to be done via an external script, but this would be a great way to populate the data easily if it were built in.

An interim solution could also be allowing the import of data from CSV sheets, but a good template should be downloadable from the app.

Alternatively, a plug-in system which would allow users to create their own implementations and share could also be a shout, especially due to separating a responsibility for maintaining the integration.

**Comment by flying-sausages** [comment]
Little too late but I see the counterarguments to a direct implementation into the app for this, so I would say a CSV import/plug-in system would be an optimal solution.

**Comment by Xtrendence** [comment]
That's a very good point. There already is a CSV import/export feature for your holdings and transactions, so you could reformat the one from your exchange to match what Cryptofolio expects in the mean time.

**Comment by Xtrendence** [comment]
If you create a few transactions with the different types on the Activity page and export, you can get a feel for what it expects.

**Comment by Xtrendence** [comment]
Added to the [project board](https://github.com/Xtrendence/Cryptofolio/projects/1#card-64055277).

**Comment by pfmephisto** [comment]
Could it be an idea to use the [ccxt](https://github.com/ccxt/ccxt) library and import data from exchanges through that?
That would probably save you from having to implement and test every single exchange yourself.

**Comment by Xtrendence** [comment]
@pfmephisto I used to have that on the project board, but removed it after I realized I'd have no way to test it. The exchange I use doesn't have read-only API tokens, so I'd have to sign up to different exchanges and buy some different types of cryptos and whatnot to be able to test stuff. In the future, if I use a different exchange with that functionality, then I'll use CCXT for that. I honestly don't trust myself to interact with an exchange API that isn't read-only where my crypto is.
